packages/node-cli/src/commands/actionresult/ActionResult.js

```diff
@@ -0,0 +1,42 @@
+/*
+** Copyright (c) 2019 Oracle and/or its affiliates.  All rights reserved.
+** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+*/
+'use strict';
+const assert = require('assert');
+const ActionResultStatus = require('../actionresult/ActionResultStatus');
+
+module.exports = class ActionResult {
+
+	constructor(options) {
+		assert(options);
+		assert(options.status, "status is required when creating an ActionResult object.");
+
+		if (options.status === ActionResultStatus.SUCCESS) {
+			assert(options.context, "context is required when ActionResult is a success.");
+		}
+
+		if (options.status === ActionResultStatus.ERROR) {
+			assert(options.error, "error is required when ActionResult is an error.");
+		}
+
+		this._status = options.status;
+		this._context = options.context;
+		this._error = options.error;
+
+		Object.preventExtensions(this);
+	}
+
+	get getStatus() {
```

Comment:
`getStatus` is very Java oriented. It should just be `status`. Same for other getters

Reply:
Do you refer to?

```
get status() {
		return this._status;
	}
```

or just to put status as a public variable?